example/android-tutorial/img-segm-demo-app/app/src/main/java/com/example/imagesegmentationdemo/MainActivity.java

```diff
@@ -0,0 +1,218 @@
+
+import com.bumptech.glide.Glide;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class MainActivity extends AppCompatActivity
+{
+    private Button btnOpenImage;
+    private Button btnCallService;
+
+    private ImageView imageView;
+
+    private ProgressBar progressBar;
+
+    private boolean isInputImageUploaded = false;
+
+    private HandlerMainActivity handler;
+
+    private SNETServiceHelper serviceHelper;
+
+    private Uri imageUri;
+
+    private String[] appPermissions;
+
+    private static final int REQUEST_CODE_OPEN_IMAGE = 0;
+    private static final int REQUEST_CODE_PERMISSIONS = 1234;
+
+    public final int maxImageHeight = 1024;
+    public final int maxImageWidth = 1024;
+
+    public static AlertDialog.Builder newAlertDialogBuilder(Context context) {
+        return new AlertDialog.Builder(context)
+                .setCancelable(false)
+                .setPositiveButton(android.R.string.yes, new DialogInterface.OnClickListener() {
+                    public void onClick(DialogInterface dialog, int which)
+                    {
+                        dialog.dismiss();
+                    }
+                })
+                .setIcon(android.R.drawable.ic_dialog_alert);
+    }
+
+    public void msgShowException(Exception e)
+    {
+        msgShowError(e.getMessage());
+    }
+
+    public void msgShowError(String message)
+    {
+        handler.sendMessage(handler.obtainMessage(
+                HandlerMainActivity.MSG_SHOW_ERROR, message));
+    }
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState)
+    {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        btnOpenImage = findViewById(R.id.btnOpenImage);
+        btnCallService= findViewById(R.id.btnCallService);
+        btnCallService.setEnabled(false);
+
+        imageView = findViewById(R.id.imageView);
+
+        progressBar = findViewById(R.id.progressBar);
+        progressBar.setVisibility(View.INVISIBLE);
+
+        handler = new HandlerMainActivity(this);
+        serviceHelper = new SNETServiceHelper(handler);
+
+        appPermissions = new String[]{
+                Manifest.permission.INTERNET,
+                Manifest.permission.READ_EXTERNAL_STORAGE,
+                Manifest.permission.WRITE_EXTERNAL_STORAGE
+        };
+
+        initialize();
+    }
+
+    public void enableActivityGUI()
+    {
+        progressBar.setVisibility(View.INVISIBLE);
+        getWindow().clearFlags(WindowManager.LayoutParams.FLAG_NOT_TOUCHABLE);
+    }
+
+    public void disableActivityGUI()
+    {
+        progressBar.setVisibility(View.VISIBLE);
+        getWindow().setFlags(WindowManager.LayoutParams.FLAG_NOT_TOUCHABLE,
+                WindowManager.LayoutParams.FLAG_NOT_TOUCHABLE);
+    }
+    public void setImageBitmap(Bitmap bm)
+    {
+        imageView.setImageBitmap(bm);
+    }
+
+    private List<String> getRequiredPermissions()
```

Comment:
I would inline this method to have all permissions related code in one code block.